### PR TITLE
Add a limit for executed jobs

### PR DESCRIPTION
**Pull request by maximemenager**
Added limits were set with personal perception, we can discuss/change them if needed.

### CODE REVIEW

File: pkg/limits/rate_limiting.go [hunk 1]
```diff
@@ -204,3 +287,21 @@ func ResetCounter(p prefixer.Prefixer, ct CounterType) {
...
+}
+
+// GetMaximumLimit returns the limit of a CounterType
+func GetMaximumLimit(ct CounterType) int64 {
+	return configs[ct].Limit
+}
+
+// SetMaximumLimit sets a new limit for a CounterType
+func SetMaximumLimit(ct CounterType, newLimit int64) {
+	mutex.Lock()
```
Comment (nono) [inline pkg/limits/rate_limiting.go]: What is the purpose of this lock?

File: model/job/mem_broker.go [hunk 2]
```diff
@@ -188,6 +190,22 @@ func (b *memBroker) PushJob(db prefixer.Prefixer, req *JobRequest) (*Job, error)
...
 	}
 
+	// Check for limits
+	ct, err := GetCounterTypeFromWorkerType(req.WorkerType)
+	if err == nil {
+		err := limits.CheckRateLimit(db, ct)
+		if err == limits.ErrRateLimitReached {
+			joblog.WithFields(logrus.Fields{
+				"worker_type": req.WorkerType,
+				"instance":    db.DomainName(),
```
Comment (nono) [inline model/job/mem_broker.go]: Can you also add the namespace like in most of the logs?

File: model/job/redis_broker.go [hunk 3]
```diff
@@ -193,6 +195,22 @@ func (b *redisBroker) PushJob(db prefixer.Prefixer, req *JobRequest) (*Job, erro
...
 	}
 
+	// Check for limits
+	ct, err := GetCounterTypeFromWorkerType(req.WorkerType)
+	if err == nil {
+		err := limits.CheckRateLimit(db, ct)
+		if err == limits.ErrRateLimitReached {
+			joblog.WithFields(logrus.Fields{
+				"worker_type": req.WorkerType,
+				"instance":    db.DomainName(),
```
Comment (nono) [inline model/job/redis_broker.go]: Idem

File: pkg/limits/rate_limiting.go [hunk 4]
```diff
@@ -77,6 +101,60 @@ var configs = []counterConfig{
...
+		Prefix: "job-service",
+		Limit:  500,
+		Period: 1 * time.Hour,
+	},
+	// JobNotification
+	{
+		Prefix: "job-push",
+		Limit:  1000,
+		Period: 1 * time.Hour,
+	},
```
Comment (nono) [inline pkg/limits/rate_limiting.go]: Let's be more strict on the limits:

- share-replicate & share-upload: 500
- konnector & service: 100
- zip: 100
- sendmail & push: 30

File: model/job/mem_broker.go [hunk 5]
```diff
@@ -196,6 +196,7 @@ func (b *memBroker) PushJob(db prefixer.Prefixer, req *JobRequest) (*Job, error)
 		err := limits.CheckRateLimit(db, ct)
 		if err == limits.ErrRateLimitReached {
 			joblog.WithFields(logrus.Fields{
+				"namespace":   "mem-broker",
```
Comment (nono) [inline model/job/mem_broker.go]: Well, it is `nspace` in fact: https://github.com/cozy/cozy-stack/blob/master/pkg/logger/logger.go#L88

Comment (maximemenager) [reply model/job/mem_broker.go]: Woops :x

However, `joblog` already has a namespace: https://github.com/cozy/cozy-stack/blob/3d3dc465738247292ca0f9657dff888f9be13aa0/model/job/broker.go#L116

Should I change it ?

Comment (nono) [reply model/job/mem_broker.go]: OK, let's keep the namespace from joblog

**Pull request merged**